saos-persistence/src/main/java/pl/edu/icm/saos/persistence/search/implementor/LawJournalEntryJpqlSearchImplementor.java

```diff
@@ -0,0 +1,78 @@
+package pl.edu.icm.saos.persistence.search.implementor;
+
+import static com.google.common.collect.Maps.newHashMap;
+
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import pl.edu.icm.saos.persistence.model.LawJournalEntry;
+import pl.edu.icm.saos.persistence.search.dto.LawJournalEntrySearchFilter;
+
+/**
+ * @author madryk
+ */
+@Service
+public class LawJournalEntryJpqlSearchImplementor extends AbstractJpqlSearchImplementor<LawJournalEntrySearchFilter, LawJournalEntry> {
+
+    
+    //------------------------ LOGIC --------------------------
+    
+    @Override
+    protected String createQuery(LawJournalEntrySearchFilter searchFilter) {
+        StringBuilder jpql = new StringBuilder(" select journalEntry from " + LawJournalEntry.class.getName() + " journalEntry ");
+        return appendConditions(searchFilter, jpql);
+    }
+
+    @Override
+    protected Map<String, Object> createParametersMap(LawJournalEntrySearchFilter searchFilter) {
+        Map<String, Object> map = newHashMap();
+        
+        if(searchFilter.getYear() != null) {
+            map.put("year", searchFilter.getYear());
+        }
+
+        if(searchFilter.getJournalNo() != null) {
+            map.put("journalNo", searchFilter.getJournalNo());
+        }
+
+        if(searchFilter.getEntry() != null) {
+            map.put("entry", searchFilter.getEntry());
+        }
+        
+        if(searchFilter.getText() != null) {
+            map.put("beginningText", searchFilter.getText() + '%');
+            map.put("middleText", "% " + searchFilter.getText() + '%');
```

Comment:
Not exacly: middleText assumes that there will be space before text.
For example lets assume that title in LawJournalEntry is "Kodeks postępowania karnego".
1) We search with text "postępowania"
 beginningText will not match
 middleText will match
2) We search with text "Kodeks"
 beginningText will match
 middleText will not match
I didn't write `'%' + searchFilter.getText() + '%'` (without space after first %) because i want to avoid match if we would type for example "ostępowania"